src/github.com/matrix-org/dendrite/common/basecomponent/base.go

```diff
@@ -78,7 +76,7 @@ func NewBaseDendrite(cfg *config.Dendrite, componentName string) *BaseDendrite {
 
 // Close implements io.Closer
 func (b *BaseDendrite) Close() error {
-	return b.tracerCloser.Close()
+	return b.tracers.Close()
```

Comment:
You call tracers.Close() twice. This is just one example
https://github.com/matrix-org/dendrite/blob/9831fcb855a793f4c3571999b32b16b251035374/src/github.com/matrix-org/dendrite/cmd/dendrite-sync-api-server/main.go#L26-L30

Reply:
Oh yes, good catch!